snapcraft/tests/test_yaml.py

```diff
@@ -426,13 +441,99 @@ def test_invalid_yaml_missing_description(self, mock_loadPlugin):
             "'description' is a required property")
 
     @unittest.mock.patch('snapcraft.internal.yaml.Config.load_plugin')
+    def test_invalid_yaml_missing_confinement(self, mock_loadPlugin):
```

Comment:
it is no longer invalid. Something like test_yaml_missing_confinement_must_log_warning.